torchbenchmark/util/framework/detectron2/model_factory.py

```diff
@@ -90,6 +90,8 @@ def __init__(self, variant, test, device, jit=False, batch_size=None, extra_args
             args.fcos_use_bn = True
 
         cfg = setup(args)
+        if hasattr(cfg, "MODEL") and cfg.MODEL.DEVICE != self.device:
```

Comment:
What happens if cfg doesn't have the "MODEL" attr?

Reply:
From the [impl](https://github.com/facebookresearch/detectron2/blob/80e2673da161f57afe37ef769836a61976108ef1/detectron2/modeling/meta_arch/build.py#L23) of buliding detectron2 models, it looks to me that the assumption is that a `cfg` should always have the attribute `MODEL`. But per the last CI run, it seems that this assumption no longer holds at least for model `test_detectron2_fcos_r_50_fpn_eval_cuda`. Details [here](https://github.com/pytorch/benchmark/actions/runs/4258041173/jobs/7420088377#step:9:271). I am not familiar with this test and wonder what you recommendation would be for this case. Thanks!

Reply:
Yes, this is because `detectron2_fcos_r_50_fpn_eval_cuda` doesn't use `MODEL.DEVICE` to initialize the model device. I am pretty sure this PR only enables the detectron2_fasterrcnn and detectron2_maskrcnn models for cpu-only torch, but [detectron2_fcos_r_50_fpn](https://github.com/pytorch/benchmark/tree/main/torchbenchmark/models/detectron2_fcos_r_50_fpn) is probably still GPU-only.

I am okay with that, and we can enable CPU with detectron2_fcos_r_50_fpn in another PR.